tensorflow_datasets/core/lazy_imports_lib.py

```diff
@@ -154,3 +225,5 @@ def test_foo(cls):
 
 
 lazy_imports = LazyImporter  # pylint: disable=invalid-name
+sys.path_hooks.append(LazyImporter)
```

Comment:
I don't think `LazyImporter` should be the same class as the hook. I would move all the hook logic inside `LazyImportHook` class